src/objectExplorer/objectExplorerService.ts

```diff
@@ -341,6 +349,9 @@ export class ObjectExplorerService {
                 this._sessionIdToPromiseMap.set(response.sessionId, promise);
                 return;
             }
+        } else {
+            // no connection was made
+            return Promise.reject();
```

Comment:
Instead of returning Promise.reject it'd be better to actually return an error.

Reply:
But there's no error here, it's just that the user pressed `Esc`. I don't think we should return an error.

Reply:
Rejecting is implying there's an error so you're already doing that. That's why you added the try/catch around the call above.

 I generally agree with you though and would suggest just returning instead and making sure that the caller handles an undefined tree node.

Reply:
It's a scenario like this https://github.com/microsoft/azuredatastudio/pull/7711/files

Unfortunately just as @Charles-Gagnon said the infrastructure here is pretty meh